akka-http-core/src/main/scala/akka/http/impl/engine/client/PoolConductor.scala

```diff
@@ -150,28 +150,28 @@ private object PoolConductor {
             case SlotEvent.ConnectedEagerly(slotIx) ⇒
             // do nothing ...
           }
-          pull(slotIn)
+          pull(slotEventsIn)
           val wasBlocked = nextSlot == -1
           nextSlot = bestSlot()
           val nowUnblocked = nextSlot != -1
-          if (wasBlocked && nowUnblocked) pull(ctxIn) // get next request context
+          if (wasBlocked && nowUnblocked) pull(requestContextsIn) // get next request context
         }
       })
 
-      setHandler(out, eagerTerminateOutput)
+      setHandler(slotCommandsOut, eagerTerminateOutput)
 
-      val tryPullCtx = () ⇒ if (nextSlot != -1 && !hasBeenPulled(ctxIn)) pull(ctxIn)
+      val tryPullCtx = () ⇒ if (nextSlot != -1 && !hasBeenPulled(requestContextsIn)) pull(requestContextsIn)
 
       override def preStart(): Unit = {
-        pull(ctxIn)
-        pull(slotIn)
+        pull(requestContextsIn)
+        pull(slotEventsIn)
 
         // eagerly start at least slotSettings.minSlots connections
         (0 until slotSettings.minSlots).foreach { connect }
       }
 
       def connect(slotIx: Int): Unit = {
-        emit(out, SwitchSlotCommand(ConnectEagerlyCommand, slotIx))
+        emit(slotCommandsOut, SwitchSlotCommand(ConnectEagerlyCommand, slotIx))
         slotStates(slotIx) = Idle
       }
```

Comment:
I would recommend leaving these changes out as they don't contribute to the fix (they're good changes but unrelated to the fix).